server/cloudformation/ecs.go

```diff
@@ -120,6 +120,12 @@ func (p *ECSServiceResource) Provision(ctx context.Context, req customresources.
 			} else {
 				err = fmt.Errorf("no primary deployment found")
 			}
+		} else {
```

Comment:
Although, I guess in this case, `d` would be nil, since the there's no PRIMARY deployment, in which case we wouldn't want to return an error.